### PR TITLE
Rewrite docu as template for cleaning up `orderings.jl`

### PR DESCRIPTION
**Pull request by wdecker**
@tthsqe12 Please:
- `wdeglex`, `wdegrevlex`, `negwdeglex`, `negwdegrevlex`: Assert entries of `W` are positive.
- `matrix_ordering(V::AbstractVector{<:MPolyElem}, M::Union{Matrix{T}, MatElem{T}})`, `matrix_ordering(R::MPolyRing, M::Union{Matrix{T}, MatElem{T}})`: Assert matrix has rank `ncols(M)` which in turn must agree with `length(V)` and `nvars(R)`, respectively.
- Introduce a function `matrix(ord::MonomialOrdering)` as described in the docu. I believe, this is what `canonical_weight_matrix(M::MonomialOrdering)` does.
- Introduce a function `weight_ordering(W::Vector{Int}, ord::MonomialOrdering)` as described in the docu. Replace the old function `weight_ordering` by the new one.
- Adjust `is_global`, `is_local`, `is_mixed` as discussed.

### CODE REVIEW

File: src/Rings/orderings.jl [hunk 1]
```diff
@@ -645,9 +848,29 @@ end
 
 @doc Markdown.doc"""
     negwdegrevlex(v::AbstractVector{<:MPolyElem}, w::Vector{Int}) -> MonomialOrdering
```
Comment (ederc) [inline src/Rings/orderings.jl]:
```suggestion
    negwdegrevlex(V::AbstractVector{<:MPolyElem}, W::Vector{Int}) -> MonomialOrdering
```

File: src/Rings/orderings.jl [hunk 2]
```diff
@@ -645,9 +848,29 @@ end
...
+    
+Given a vector `V` of variables and a vector `W` of positive integers, return the corresponding negative
+weighted reverse lexicographical ordering on the set of monomials in the given variables.
+
     negwdegrevlex(R::MPolyRing, w::Vector{Int}) -> MonomialOrdering
```
Comment (ederc) [inline src/Rings/orderings.jl]:
```suggestion
    negwdegrevlex(R::MPolyRing, W::Vector{Int}) -> MonomialOrdering
```

**Comment by wdecker** [comment]
I get the following error

 [ Info: Expanding citation: ConwaySloane99.
ERROR: LoadError: Citation not found in bibliography: ConwaySloane99
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:33
  [2] expand_citation(link::Markdown.Link, meta::Dict{Symbol, Any}, page::Documenter.Documents.Page, doc::Documenter.Documents.Document)

Can someone please adjust this?

**Comment by thofma** [comment]
Will be fixed by https://github.com/oscar-system/Oscar.jl/pull/1604.

**Comment by wdecker** [comment]
@thofma thx! There are also these errors:

┌ Error: no docs found for 'short_vectors(L::ZLat, ub)' in `@docs` block in src/Hecke/quad_forms/integer_lattices.md:161-166
│ ```@docs
│ short_vectors(L::ZLat, ub)
│ shortest_vectors(L::ZLat)
│ minimum(L::ZLat)
│ kissing_number(L::ZLat)
│ ```
└ @ Documenter.Expanders ~/.julia/packages/Documenter/yf96B/src/Utilities/Utilities.jl:32
┌ Error: no docs found for 'shortest_vectors(L::ZLat)' in `@docs` block in src/Hecke/quad_forms/integer_lattices.md:161-166
│ ```@docs
│ short_vectors(L::ZLat, ub)
│ shortest_vectors(L::ZLat)
│ minimum(L::ZLat)
│ kissing_number(L::ZLat)

┌ Error: no doc found for reference '[`automorphisms(::Hecke.NumFieldMor)`](@ref)' in src/Experimental/galois.md.

**Comment by thofma** [comment]
Yes, will also be addressed in that PR.

**Review by ederc — APPROVED** [review]

**Pull request merged**